integrated_channels/integrated_channel/tasks.py

```diff
@@ -22,6 +25,34 @@
 User = auth.get_user_model()
 
 
+def locked(expiry_seconds, lock_name_kwargs):  # lint-amnesty, pylint: disable=missing-function-docstring
+    def task_decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):  # lint-amnesty, pylint: disable=inconsistent-return-statements
+            cache_key = f'{func.__name__}'
+            for key in lock_name_kwargs:
+                cache_key = f'{cache_key}-{key}:{kwargs.get(key)}'
```

Comment:
This is over-writing `cache_key` on each pass of the loop.  Did you want something like this instead?
```python
cache_key += f'-{key}:{kwargs.get(key)}'